addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3360,6 +3360,7 @@ export class OdooEditor extends EventTarget {
                     // front of a contentEditable="false" (eg: font awesome).
                     ev.preventDefault();
                     this._applyCommand('oDeleteBackward');
+                    $(ev.target).trigger("input");
```

Comment:
No jquery please.
You can listen the event `historyStep` of the editor in `html_field.js`.
In the method `startWysiwyg`, you can add:
```js
this.wysiwyg.odooEditor.addEventListener('startWysiwyg', () => this.props.setDirty(this._isDirty()));
```
No need to remove the listener as the editor is destroy when the wysiwyg is destroyed.
No need to react on editable input as it will produce an `historyStep`.